### PR TITLE
Use urllib functions to normalize URL

### PR DESCRIPTION
**Pull request by greenscientist**
We add a normalize_url function that remove extra characters from an URL and also add build_url function so that we don't manipulate URL as string everywhere

Add a test with an extra "/" at the end of the base URL

Created with the help of chatgpt

### CODE REVIEW

**Comment by greenscientist** [comment]
@tahini , this could be better than #948

File: pyTransition/pyTransition/transition.py [hunk 1]
```diff
@@ -39,6 +42,46 @@ def __init__(self, url, username, password, token=None):
...
+    def __normalize_url(url):
+        """Normalize the URL.
+
+        Remove extra characters from the URL
+        """
+        parsed_url = urlparse(url)
+        netloc = parsed_url.netloc.lower()
+
+        if netloc.endswith(':80') and parsed_url.scheme == 'http':
+            netloc = netloc[:-3]
```
Comment (tahini) [inline pyTransition/pyTransition/transition.py:55]: What does this do? What's `:-3`?

File: pyTransition/pyTransition/transition_urlparse.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+from urllib.parse import urlparse, urlunparse, urlencode, parse_qs
+
+class Transition:
```
Comment (tahini) [inline pyTransition/pyTransition/transition_urlparse.py:3]: Remove this file?

**Review by tahini — CHANGES_REQUESTED** [review]
You seem to have committed a copy of the file, unless I'm mistaken

File: pyTransition/pyTransition/transition.py (hunk 1 above)
Comment (greenscientist) [reply pyTransition/pyTransition/transition.py:55]: You'll have to ask chatgpt... ;)

Just kidding, it remove the extra :80 if we already specified http as protocol.

**Review by tahini — APPROVED** [review]
c'est bon! j'imagine que tu as testé (je vois que le CI ne roule pas encore les tests python)

**Comment by tahini** [comment]
@greenscientist Est-ce que pip a été mis à jour avec cette nouvelle version ou il faut faire qqc manuellement?

**Comment by greenscientist** [comment]
Pas encore, je dois faire un release. 
Aujourd'hui, ou demain.

**Pull request merged**